src/reducer.ts

```diff
@@ -108,50 +139,120 @@ type ReducerCreator = {
+const SEED_ORIGIN = Symbol('Seed origin');
+
 /**
- * Combine reducer entries into a stream operator
+ * Combine registered reducers into a stream operator
  *
- * The payload of each incoming action is applied to the matching reducers
- * together with the previous state (or the seed if it's the first invocation),
- * and the returned state is emitted.
+ * Each reducer will receive the previous state (or the seed if it's the first
+ * invocation) together with the payloads of the actions of the given reducer,
+ * or the emitted values from the stream of the given reducer.
  *
- * This operator does not change whether the stream is hot or cold.
+ * The behaviour is undefined if multiple reducers are registered for the same
+ * actions.
+ *
+ * This operator will only emit on relevant actions or if one of the source
+ * streams emits. Specifically, if the seed is a stream, this operator will not
+ * emit when the seed stream emits, but the first time an action of source
+ * stream emits after the seed stream has emitted. The seed stream will be
+ * unsubscribed after its first emit.
  *
  * @param seed The initial input to the first reducer call
  * @param reducers The reducer entries that should be combined
  */
 export const combineReducers = <State>(
-  seed: State,
+  seed: State | Observable<State>,
   reducers: RegisteredReducer<State, any>[]
 ): OperatorFunction<UnknownAction, State> => {
+  const actionReducers = reducers.filter(isActionReducer);
+  const streamReducers = reducers.filter(isStreamReducer);
   const reducersByActionType = new Map(
-    reducers.flatMap(reducerFn =>
+    actionReducers.flatMap(reducerFn =>
       reducerFn.trigger.actions.map(action => [action.type, reducerFn])
     )
   );
+
+  type ScanInput =
+    | readonly [typeof SEED_ORIGIN, State]
+    | readonly [typeof ACTION_ORIGIN, UnknownAction]
+    | readonly [number, any];
+  type InitializedAccumulator = {
+    initialized: true;
+    reduced: boolean;
+    state: State;
+  };
+  type Accumulator = { initialized: false } | InitializedAccumulator;
+
+  const seed$ = wrapInObservable(seed).pipe(
+    take(1),
+    map(payload => [SEED_ORIGIN, payload] as ScanInput)
+  );
+  const source$s = streamReducers.map((reducerFn, i) =>
+    reducerFn.trigger.source$.pipe(map(payload => [i, payload] as ScanInput))
+  );
+
   return pipe(
-    ofType(...reducers.flatMap(reducerFn => reducerFn.trigger.actions)),
-    scan((state, { type, payload }: UnknownAction) => {
-      const RegisteredReducer = reducersByActionType.get(type);
-      if (RegisteredReducer === undefined) {
-        // This shouldn't be possible
-        return state;
-      }
+    ofType(...actionReducers.flatMap(reducerFn => reducerFn.trigger.actions)),
+    map(action => [ACTION_ORIGIN, action] as ScanInput),
+    merge(seed$, ...source$s),
+    scan(
+      (accumulator, packet) => {
+        if (packet[0] === SEED_ORIGIN) {
+          return { initialized: true, reduced: false, state: packet[1] };
```

Comment:
If if this is internal I think it would make sense to destruct packet to named variables